client/dashboard/sites/settings-site-visibility/launch-form.tsx

```diff
@@ -0,0 +1,31 @@
+import {
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import type { Site } from '../../data/types';
+
+export function LaunchForm( { site }: { site: Site } ) {
+	const handleLaunch = () => {
+		window.location.href = addQueryArgs( '/start/launch-site', {
+			siteSlug: site.slug,
+			new: site.name,
+			hide_initial_query: 'yes',
+		} );
+	};
+
+	return (
+		<VStack spacing={ 4 } alignment="left">
+			<Text>
+				{ __(
+					'Your site hasn\'t been launched yet. It is hidden from visitors behind a "Coming Soon" notice until it is launched.'
+				) }
+			</Text>
+			<Button __next40pxDefaultSize variant="primary" onClick={ handleLaunch }>
```

Comment:
why not an `href` directly instead of `onClick` it's better a11y wise.

Reply:
The one already implemented is for basic cases only. We haven't ported more complex cases yet:

https://github.com/Automattic/wp-calypso/blob/d102f9f5e7a23bb265b0e501b2d51016a8b0acbc/client/sites/settings/site/visibility/index.jsx#L125-L133

Reply:
It feels the complex case should be handled separately, if we want to show the modal, we use onClick, if we want a link we use href.

Reply:
Makes sense; done.